src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMCodegenCompilation.CodeGen.cs

```diff
@@ -47,6 +48,116 @@ public override ISymbolNode GetExternalMethodAccessor(MethodDesc method, ReadOnl
 
         public override CorInfoLlvmEHModel GetLlvmExceptionHandlingModel() => Options.ExceptionHandlingModel;
 
+        internal WasmFunctionType GetWasmFunctionTypeForMethod(MethodSignature signature, bool hasHiddenParam)
+        {
+            WasmValueType wasmPointerType = _nodeFactory.Target.PointerSize == 4 ? WasmValueType.I32 : WasmValueType.I64;
+
+            WasmValueType GetWasmTypeForTypeDesc(TypeDesc type)
```

Comment:
Just curious, has struct scalarization (from the WASM C ABI) been performed at this point? Or do you have to do it?

Reply:
Yes, it has - all of the callers of this function first check `GetPrimitiveTypeForTrivialWasmStruct`. So here we only expect primitives.

Reply:
Is this related to multi value return types or does LLVM never produce those?

Reply:
https://webassembly.org/features/  suggests it is standardized, but if we don't support it then no problem.